resonate/bridge.py

```diff
@@ -440,9 +440,15 @@ def _handle_network_request(self, cmd_id: str, cid: str, req: CreatePromiseReq |
             case _:
                 raise NotImplementedError
 
-    def _handle_function(self, func: Callable[[], Any]) -> Result:
+    def _validate[T](self, v: T, validation: Callable[[T], bool] | None) -> T:
+        if validation is not None and not validation(v):
+            msg = "function result didn't passed validation"
+            raise ResonateValidationError(msg)
+        return v
+
+    def _handle_function(self, func: Callable[[], Any], validation: Callable[[Any], bool] | None) -> Result:
         try:
-            r = func()
+            r = self._validate(func(), validation)
```

Comment:
It looks like we are doing this here and in the scheduler, can we consolidate into one place? My inclination would be to do it in the scheduler